City-of-Helsinki/drupal-helfi-sote
# UHF-5512: Sote unit contact card

# Add unit contact card [UHF-5512](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5512)
Added unit contact card paragraph which shows referenced units.

## What was done
* Added unit contact card paragraph type.
  * Paragraph includes fields for selecting which unit fields are shown.
* The paragraph shows a referenced unit, which can be published or not.
* Enabled the new paragraph for landing page, page, unit page and service page.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin UHF-5512-sote-unit-contact-card`
* Update the Helfi TPR module and import TPR units.
  * Check instructions from https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/122
* Run `make drush-cim`
* Run `make drush-cr`

## How to test
* Make sure you can import highlight content to TPR units (see the attached PR for more info).
* Make sure you can add _unit contact card_ paragraphs to landing pages, basic pages (lower region), TPR units (lower region) and TPR services (lower region).
* Make sure that when selecting what information is shown (from the paragraph), the correct information is shown when viewing the element.
* Make sure the feature works with both published and unpublished units. It should be possible to show information also from unpublished units. With unpublished units the link to the unit's page is not shown.
* Make sure the UI matches the design (the link to the design file is at the issue page).

## Other PRs
* https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/122
